### ISSUE TITLE
Support providing `name` for application when deploying

### ISSUE THREAD

**Issue by mikkelhegn** [fermyon/cloud-plugin#8]
The cloud api supports defining a `name` for an application, when creating it. Today this is inferred from the `spin.toml` file on deployment.

It would be great to be able to define this as part of deployment. E.g. `spin cloud deploy --name my-app`

**Comment by lann** [comment]
> The behavior today to create a new app, would be to change the name in spin.toml, and I think it's a bit weird.

Yeah I think this is the underlying issue. There is nothing else in the manifest you _have_ to change as an operational detail (at least once we have cloud config).
